java/src/main/java/com/powsybl/dataframe/network/adders/CurveReactiveLimitsDataframeAdder.java

```diff
@@ -0,0 +1,152 @@
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.ReactiveCapabilityCurveAdder;
+
+/**
+ * @author Massimo Ferraro <massimo.ferraro@soft.it>
+ */
+public class CurveReactiveLimitsDataframeAdder implements NetworkElementAdder {
+
+    private static final List<SeriesMetadata> METADATA = List.of(
+            SeriesMetadata.stringIndex("id"),
+            SeriesMetadata.intIndex("num"),
+            SeriesMetadata.doubles("p"),
+            SeriesMetadata.doubles("min_q"),
+            SeriesMetadata.doubles("max_q")
+    );
+
+    private static final class CurveReactiveLimitsSeries {
+
+        private final StringSeries elementIds;
+        private final IntSeries nums;
+        private final DoubleSeries ps;
+        private final DoubleSeries minQs;
+        private final DoubleSeries maxQs;
+
+        CurveReactiveLimitsSeries(UpdatingDataframe dataframe) {
+            this.elementIds = getRequiredStrings(dataframe, "id");
+            this.nums = getRequiredInts(dataframe, "num");
+            this.ps = getRequiredDoubles(dataframe, "p");
+            this.minQs = getRequiredDoubles(dataframe, "min_q");
+            this.maxQs = getRequiredDoubles(dataframe, "max_q");
+        }
+
+        public StringSeries getElementIds() {
+            return elementIds;
+        }
+
+        public IntSeries getNums() {
+            return nums;
+        }
+
+        public DoubleSeries getPs() {
+            return ps;
+        }
+
+        public DoubleSeries getMinQs() {
+            return minQs;
+        }
+
+        public DoubleSeries getMaxQs() {
+            return maxQs;
+        }
+    }
+
+    private static final class CurvePoint {
+
+        private final Integer num;
+        private final Double p;
+        private final Double minQ;
+        private final Double maxQ;
+
+        CurvePoint(Integer num, Double p, Double minQ, Double maxQ) {
+            this.num = num;
+            this.p = p;
+            this.minQ = minQ;
+            this.maxQ = maxQ;
+        }
+
+        public Integer getNum() {
+            return num;
+        }
+
+        public Double getP() {
+            return p;
+        }
+
+        public Double getMinQ() {
+            return minQ;
+        }
+
+        public Double getMaxQ() {
+            return maxQ;
+        }
+
+    }
+
+    @Override
+    public List<List<SeriesMetadata>> getMetadata() {
+        return Collections.singletonList(METADATA);
+    }
+
+    @Override
+    public void addElements(Network network, List<UpdatingDataframe> dataframes) {
+        UpdatingDataframe primaryTable = dataframes.get(0);
+        CurveReactiveLimitsSeries series = new CurveReactiveLimitsSeries(primaryTable);
+        Map<String, SortedSet<CurvePoint>> curvesPoints = new HashMap<>();
+        for (int i = 0; i < primaryTable.getRowCount(); i++) {
+            String elementId = series.getElementIds().get(i);
+            Integer num = series.getNums().get(i);

```

Comment:
I think that we could spare the user the necessity to define the number for points:
we could just add them in the order of the table ?

Actually, in the adder implementation, the order is not even used: points will be sorted by P. (it's not really specified in the interface though!)

Also, in similar tables (shunts sections and transfo steps), it's done this way, so it would be more consistent.

Reply:
column 'num', for points, removed. Indeed it was redundant. Thanks.